### PR TITLE
This allows a user to specify an option of what level of issue and above should fail the build

### PR DESCRIPTION
**Pull request by jgrodskiy-donlen**
Issue #16

### CODE REVIEW

**Comment by jgrodskiy-donlen** [comment]
image

File: src/main/scala/ch/mibex/bitbucket/sonar/SonarBBPlugin.scala [hunk 1]
```diff
@@ -116,6 +117,13 @@ object SonarBBPlugin {
       description = "If enabled, the plug-in will update the build status of the pull request depending on the " +
         "Sonar analysis result. The analysis and also the build is failed if there are any critical or blocker issues.",
       global = true
+    ),
+    new Property(
+      key = SonarBBPlugin.SonarApprovalSeverityLevel,
+      name = "The issue level to look for to determine to Auto-Approve",
+      defaultValue = Severity.CRITICAL,
+      description = "If any issues of this level or higher are found, it will un approve a PR ",
```
Comment (nfalco79) [inline src/main/scala/ch/mibex/bitbucket/sonar/SonarBBPlugin.scala]: the description should be end with point and without trailing spaces.

File: src/main/scala/ch/mibex/bitbucket/sonar/review/PullRequestReviewResults.scala [hunk 2]
```diff
@@ -49,10 +52,19 @@ class PullRequestReviewResults(pluginConfiguration: SonarBBPluginConfig) {
...
-    // there does not seam to be a way to check the quality gates in preview mode, so we make an assumption about
-    // what users would consider good quality here :-)
-    newIssuesBySeverity(Severity.CRITICAL) == 0 && newIssuesBySeverity(Severity.BLOCKER) == 0
+  def canBeApproved: Boolean = {
+    val levelString = pluginConfiguration.sonarApprovalSeverityLevel();
+    val level = Severity.valueOf(levelString);
+    val allLevels = Severity.values();
+    val index = allLevels.indexOf(level);
+
+    newIssuesBySeverity.keys.filter((severity)=> {
```
Comment (nfalco79) [inline src/main/scala/ch/mibex/bitbucket/sonar/review/PullRequestReviewResults.scala]: In this portion of code miss code formatting

**Comment by nfalco79** [comment]
Good PR even if I prefer a more significant name and in accordance with the other existing ones like BitbucketApproveMinSeverity (sonar.bitbucket.approvalMinSeverity).

File: src/main/scala/ch/mibex/bitbucket/sonar/SonarBBPlugin.scala [hunk 3]
```diff
@@ -116,6 +117,13 @@ object SonarBBPlugin {
       description = "If enabled, the plug-in will update the build status of the pull request depending on the " +
         "Sonar analysis result. The analysis and also the build is failed if there are any critical or blocker issues.",
       global = true
+    ),
+    new Property(
+      key = SonarBBPlugin.SonarApprovalSeverityLevel,
```
Comment (nfalco79) [inline src/main/scala/ch/mibex/bitbucket/sonar/SonarBBPlugin.scala]: name and description of `SonarBBPlugin.BitbucketApproveUnapprove` property should be corrected according to this new behavior.

**Comment by mrueegg** [comment]
@jgrodskiy-donlen  Thanks for this pull request! I agree with the comments made by @nfalco79. As soon as the remaining naming/formatting issues are fixed, I'm happy to merge this. Thanks in advance

**Comment by nfalco79** [comment]
also miss unit tests

**Comment by mrueegg** [comment]
Fixed and merged. Thanks for your contribution!

**Pull request merged**